homeassistant/components/tuya/cover.py

```diff
@@ -235,12 +235,23 @@ def current_cover_position(self) -> int | None:
         if self._current_position_type is None:
             return None
 
-        if not (
-            dpcode := (
+        # Determine current_position DPCodes
+        if isinstance(self.entity_description.current_position, DPCode):
```

Comment:
This should not be determined every single call over and over again. Instead, we should determine the code to use when setting up the entity.

Reply:
i've move the code to init, and add extra variable _position_dpcode